### ISSUE TITLE
MDCDrawer issue on iOS 11.2.1

### ISSUE THREAD

**Issue by luspav** [material-components/material-components-web#1790]
material/drawer@0.27.0 down to 0.5.6

Safari mobile
iOS 11.2.1

I found this in my React-based app wrapped into cordova
but it can be easily reproduced on demo page: [https://material-components-web.appspot.com/drawer/temporary-drawer.html](https://material-components-web.appspot.com/drawer/temporary-drawer.html)

it appeared only after update to iOS 11.2.1, also can be reproduced in Simulator with iOS 11.2+

The bug:
When trying to close the drawer a simple click on the area outside or with `... mdcDrawerInstance.open = false;...` or with `mdcDrawerInstance.foundation_.close()` or any other - the drawer starts to shake and not closing

Running into small debug I found that it is caused by two things actually:

1. Some problem with animation function
Removing bezier curve and moving to simple ease - resolves the issue with shaking, but the drawer is not closing at all then
A sample workaround:
```
.mdc-temporary-drawer {
  &--animating {
    &::before {
      transition: opacity .3s ease-out;
    }

    &.mdc-temporary-drawer--open .mdc-temporary-drawer__drawer {
      transition: transform .3s ease-out;
    }

    .mdc-temporary-drawer__drawer {
      transition: none;
    }
  }
}
```

2. And the `touch end` event starts a loop of open/close function
The second workaround:
```
this._mdcTemporaryDrawer = new MDCTemporaryDrawer(this._temporaryDrawer);

this._mdcTemporaryDrawer.foundation_.handleTouchMove_ = () => {};
this._mdcTemporaryDrawer.foundation_.componentTouchMoveHandler_ = () => {};
this._mdcTemporaryDrawer.foundation_.handleTouchEnd_ = () => {
      this._mdcTemporaryDrawer.foundation_.prepareForTouchEnd_();
};
this._mdcTemporaryDrawer.foundation_.componentTouchEndHandler_ = () => {};
```

**Comment by kfranqueiro** [comment]
I think this is a duplicate of #1700, but thanks for the additional useful information!

**Comment by luspav** [comment]
@kfranqueiro yes, haven't found it. thanks

**Comment by canaanites** [comment]
@luspav where did you add this code:

```
this._mdcTemporaryDrawer = new MDCTemporaryDrawer(this._temporaryDrawer);

this._mdcTemporaryDrawer.foundation_.handleTouchMove_ = () => {};
this._mdcTemporaryDrawer.foundation_.componentTouchMoveHandler_ = () => {};
this._mdcTemporaryDrawer.foundation_.handleTouchEnd_ = () => {
      this._mdcTemporaryDrawer.foundation_.prepareForTouchEnd_();
};
this._mdcTemporaryDrawer.foundation_.componentTouchEndHandler_ = () => {};
```

**Comment by luspav** [comment]
@canaanites it is a part of my react component

```
_initMDCTemporaryDrawer() {
    if (!this._temporaryDrawer) return;

    if (!this._mdcTemporaryDrawer) {
      this._mdcTemporaryDrawer = new MDCTemporaryDrawer(this._temporaryDrawer);

      this._mdcTemporaryDrawer.foundation_.handleTouchMove_ = () => {};
      this._mdcTemporaryDrawer.foundation_.componentTouchMoveHandler_ = () => {};
      this._mdcTemporaryDrawer.foundation_.handleTouchEnd_ = () => {
        this._mdcTemporaryDrawer.foundation_.prepareForTouchEnd_();
      };
      this._mdcTemporaryDrawer.foundation_.componentTouchEndHandler_ = () => {};
    }
  }
```

```
componentDidMount() {
    this._initMDCTemporaryDrawer();
    .........
  }
```

```
render() {
   ............

    return (
      <div>
        <aside className={classnames(['mdc-temporary-drawer', ......])}
               ref={ref => (this._temporaryDrawer = ref)}>
          <nav className="mdc-temporary-drawer__drawer">
         ............
```
